### DISCUSSION TITLE
Question about how Redshift Destination works

### DISCUSSION THREAD

**Discussion by morA123R** [airbytehq/airbyte#32671] [Connector Questions]
Hi,
I wanted to ask a few questions about the Redshift destination, and hopefully I can get some answers about it:
1. I have several sources that I want to connect to the same redshift destination, and sync them in an incremental  append+deduped mode. Does each connection that I create will open a new connection to Redshift, or will it use the same connection? I would like to understand if I need to handle the rate limit of concurrent open connections.
2. I saw that Airbyte automatically creates a destination table in Redshift according to the source schema, and I saw that it defined the SORTKEY to ( _airbyte_unique_key, _airbyte_emitted_at ). If I want to change it to AUTO for better optimization, since I also query the data, could it affect the way Airbyte works?
3. If I have a source that is scheduled to sync to Redshift every hour, and the first increment takes 2 hours, and the second increment takes 30min so it can finish before the first one. Will I still get the right order of the increments in the destination table?

Thanks in advance.

**Comment by marcosmarxm** [comment]
1. Each Airbyte connection will open a new connection in Redshift.
2. The SORTKEY is used by normalization which is a deprecated module used by destinations. You can extract the dbt project and change as your will, see [here](https://docs.airbyte.com/operator-guides/transformation-and-normalization/transformations-with-sql).
3. Airbyte will have only one sync running at the time. If the first didn't finish Airbyte will schedule but not trigger the next sync.

**Reply by morA123R** [reply]
Thanks @marcosmarxm for the explanation. 
1. So if I want to handle the rate limit of concurrent open connections to the redshift, can I use the MAX_SYNC_WORKERS configuration to achieve that? If so, where can I change this configuration?
2. Is the SORTKEY not used for syncing the increments as well? And regarding the custom dbt project, can you elaborate on it? Can I use it to tell Airbye how to create the destination table, instead of what Airbyte automatically creates? which means I will be able to change the SORTKEY and also some of the data types if I want.
But if it runs in every sync, will it rerun the creation table sql script that I wanted to change, and can override the destination table?
Does the dbt project is used only for data transformation using sql, or can it run also a python code, for example?
4. Great, thank you.

[a comment by an automated account is not shown]
